universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -69,18 +69,34 @@ public ProgramDefinition updateProgramDefinition(long programId, String name, St
   @Transactional
   public ProgramDefinition addBlockToProgram(
       long programId, String blockName, String blockDescription) throws ProgramNotFoundException {
+    return addBlockToProgram(programId, blockName, blockDescription, ImmutableList.of());
+  }
+
+  @Override
+  @Transactional
+  public ProgramDefinition addBlockToProgram(
+      long programId,
+      String blockName,
+      String blockDescription,
+      ImmutableList<QuestionDefinition> questionDefinitions)
+      throws ProgramNotFoundException {
     ProgramDefinition programDefinition = getProgramOrThrow(programId);
     long blockId = getNextBlockId(programDefinition);
 
-    BlockDefinition blockDefinition =
+    BlockDefinition.Builder blockDefinitionBuilder =
         BlockDefinition.builder()
             .setId(blockId)
             .setName(blockName)
-            .setDescription(blockDescription)
-            .build();
+            .setDescription(blockDescription);
+    if (!questionDefinitions.isEmpty()) {
```

Comment:
nit: does it matter if `questionDefinitions` is empty or not? Couldn't we `setQuestionDefinitions(questionDefinitions)` either way?

Reply:
thank you for calling that out, fixed!